nanopub/profile.py

```diff
@@ -62,36 +63,31 @@ def __init__(
         else:
             self._private_key = private_key
 
-        if not public_key:
+        if not public_key and private_key:
             log.info('Public key not provided when loading the Nanopub profile, generating it from the provided private key')
-            key = RSA.importKey(decodebytes(self._private_key.encode()))
-            self._public_key = key.publickey().export_key().decode('utf-8')
-        else:
-            if isinstance(public_key, Path):
-                try:
-                    with open(public_key) as f:
-                        self._public_key = f.read().strip()
-                except FileNotFoundError:
-                    raise ProfileError(
-                        f'Private key file {public_key} for nanopub not found.\n'
-                        f'Maybe your nanopub profile was not set up yet or not set up '
-                        f'correctly. \n{PROFILE_INSTRUCTIONS_MESSAGE}'
-                    )
-            else:
-                self._public_key = public_key
-
+            key = RSA.import_key(decodebytes(self._private_key.encode()))
+            self._public_key = format_key(key.publickey().export_key().decode('utf-8'))
+        elif isinstance(public_key, Path):
+            try:
+                with open(public_key) as f:
+                    self._public_key = f.read().strip()
+            except FileNotFoundError:
+                raise ProfileError(
+                    f'Private key file {public_key} for nanopub not found.\n'
+                    f'Maybe your nanopub profile was not set up yet or not set up '
+                    f'correctly. \n{PROFILE_INSTRUCTIONS_MESSAGE}'
+                )
+        elif public_key:
+            self._public_key = public_key
 
     def generate_keys(self) -> str:
         """Generate private/public RSA key pair at the path specified in the profile.yml, to be used to sign nanopubs"""
         key = RSA.generate(2048)
```

Comment:
In principle you should be able to use other non-RSA encryption, not sure if the nanopub tooling already supports it though (e.g. can you publish a nanopub signed using ECDSA?). We would need to ask @tkuhn 

But in general it makes sense to clearly define the encryption configuration somewhere it is easy to access instead of hiding it in the code, and since we use it at 2 different places it would be better to have the size defined only once
So I created `RSA_KEY_SIZE` in `definitions.py`